library/src/tests/intrinsic.rs

```diff
@@ -1832,38 +1832,38 @@ fn test_base_mctadj_4_control() {
     );
     expect![[r#"
         STATE:
-        |0000000000⟩: 0.1734+0.0345𝑖
-        |0000100001⟩: 0.1734+0.0345𝑖
-        |0001000010⟩: 0.1734+0.0345𝑖
-        |0001100011⟩: 0.1734+0.0345𝑖
-        |0010000100⟩: 0.1734+0.0345𝑖
-        |0010100101⟩: 0.1734+0.0345𝑖
-        |0011000110⟩: 0.1734+0.0345𝑖
-        |0011100111⟩: 0.1734+0.0345𝑖
-        |0100001000⟩: 0.1734+0.0345𝑖
-        |0100101001⟩: 0.1734+0.0345𝑖
-        |0101001010⟩: 0.1734+0.0345𝑖
-        |0101101011⟩: 0.1734+0.0345𝑖
-        |0110001100⟩: 0.1734+0.0345𝑖
-        |0110101101⟩: 0.1734+0.0345𝑖
-        |0111001110⟩: 0.1734+0.0345𝑖
-        |0111101111⟩: 0.1734+0.0345𝑖
-        |1000010000⟩: 0.1734+0.0345𝑖
-        |1000110001⟩: 0.1734+0.0345𝑖
-        |1001010010⟩: 0.1734+0.0345𝑖
-        |1001110011⟩: 0.1734+0.0345𝑖
-        |1010010100⟩: 0.1734+0.0345𝑖
-        |1010110101⟩: 0.1734+0.0345𝑖
-        |1011010110⟩: 0.1734+0.0345𝑖
-        |1011110111⟩: 0.1734+0.0345𝑖
-        |1100011000⟩: 0.1734+0.0345𝑖
-        |1100111001⟩: 0.1734+0.0345𝑖
-        |1101011010⟩: 0.1734+0.0345𝑖
-        |1101111011⟩: 0.1734+0.0345𝑖
-        |1110011100⟩: 0.1734+0.0345𝑖
-        |1110111101⟩: 0.1734+0.0345𝑖
-        |1111011110⟩: 0.1734+0.0345𝑖
-        |1111111111⟩: 0.1470−0.0982𝑖
+        |0000000000⟩: 0.1768+0.0000𝑖
+        |0000100001⟩: 0.1768+0.0000𝑖
+        |0001000010⟩: 0.1768+0.0000𝑖
+        |0001100011⟩: 0.1768+0.0000𝑖
+        |0010000100⟩: 0.1768+0.0000𝑖
+        |0010100101⟩: 0.1768+0.0000𝑖
+        |0011000110⟩: 0.1768+0.0000𝑖
+        |0011100111⟩: 0.1768+0.0000𝑖
+        |0100001000⟩: 0.1768+0.0000𝑖
+        |0100101001⟩: 0.1768+0.0000𝑖
+        |0101001010⟩: 0.1768+0.0000𝑖
+        |0101101011⟩: 0.1768+0.0000𝑖
+        |0110001100⟩: 0.1768+0.0000𝑖
+        |0110101101⟩: 0.1768+0.0000𝑖
+        |0111001110⟩: 0.1768+0.0000𝑖
+        |0111101111⟩: 0.1768+0.0000𝑖
+        |1000010000⟩: 0.1768+0.0000𝑖
+        |1000110001⟩: 0.1768+0.0000𝑖
+        |1001010010⟩: 0.1768+0.0000𝑖
+        |1001110011⟩: 0.1768+0.0000𝑖
+        |1010010100⟩: 0.1768+0.0000𝑖
+        |1010110101⟩: 0.1768+0.0000𝑖
+        |1011010110⟩: 0.1768+0.0000𝑖
+        |1011110111⟩: 0.1768+0.0000𝑖
+        |1100011000⟩: 0.1768+0.0000𝑖
+        |1100111001⟩: 0.1768+0.0000𝑖
+        |1101011010⟩: 0.1768+0.0000𝑖
+        |1101111011⟩: 0.1768+0.0000𝑖
+        |1110011100⟩: 0.1768+0.0000𝑖
+        |1110111101⟩: 0.1768+0.0000𝑖
+        |1111011110⟩: 0.1768+0.0000𝑖
+        |1111111111⟩: 0.1250−0.1250𝑖
```

Comment:
That's a LOT of expect tests that were previously 'wrong'. How did we not catch this before? Or did we just not worry about it in these cases before for some reason?

Reply:
Yeah, it's an embarrassing oversight on my part. Because we have other tests on smaller states for S that looked correct, I assumed these were correct and didn't dig into the state. But even still, controlled-T being off should have been easier to catch.

One thing we can consider, now that we have DumpOperation support, is converting these test to be based on the matrix rather than the state, which is easier to verify both for the original gate and for controls of any size (since it's just diagonal ones for the additional qubits). If you think it's worth the churn we could do that conversion here, if not we can save it for another test focused PR.

Reply:
No problem. Thanks for the details. I'll leave it up to you as to if/when the changes to validation make sense.